pegasus/sites.v3/hourofcode.com/public/js/signup-form.js

```diff
@@ -0,0 +1,163 @@
+    if (($('#hoc-event-type').val() === 'in_school') && ($("#country").val() === 'US')) {
+      $('#school-name-field').show();
+      $('#organization-name-field').hide();
+      $('#hoc-event-location-field').show();
+      $('#hoc-entire-school').show();
+      // continue button goes to census questions on click
+      $('.continue-btn').show();
+      $('#submit').hide();
+    } else if (($('#hoc-event-type').val() === 'in_school')){
+    // in-school & NOT US
+      $('#school-name-field').show();
+      $('#organization-name-field').hide();
+      $('#hoc-event-location-field').show();
+      $('#hoc-entire-school').show();
+      $('.continue-btn').hide();
+      $('#submit').show();
+    } else if ($('#hoc-event-type').val() === 'out_of_school') {
+      // out of school, either US or non-US
+      $('#organization-name-field').show();
+      $('#hoc-event-location-field').show();
+      $('#school-name-field').hide();
+      $('#hoc-entire-school').hide();
+      $('.continue-btn').hide();
+      $('#submit').show();
+    }
+  }
+
+  $('#country').change(function () {
+    checkShowNameEventLocation();
+  });
+
+  $('#hoc-event-type').change(function () {
+    checkShowNameEventLocation();
+  });
+
+  function checkShowCensusFollowUp() {
+    if ($("#twenty-hour-how-much").val() === "some" || $("#twenty-hour-how-much").val() === "all" || $("#ten-hour-how-much").val() === "some" ||
+    $("#ten-hour-how-much").val() === "all") {
+      $('#followup_questions').show();
+    } else {
+      $('#followup_questions').hide();
+    }
+  }
+
+  $('#twenty-hour-how-much').change(function () {
+    checkShowCensusFollowUp();
+  });
+
+  $('#ten-hour-how-much').change(function () {
+    checkShowCensusFollowUp();
+  });
+
+  $('#role').change(function () {
+    if ($(this).val() === "teacher" || $(this).val() === "administrator") {
+      $('#pledge').show();
+    } else {
+      $('#pledge').hide();
+    }
+  });
+ });
+
+function signupFormComplete(data) {
+  window.location = "#{resolve_url('thanks')}";
+}
+
+function validateFields() {
+  if ($("#hoc-name").val() === "") {
+    $('#name-error').show();
+    return false;
+  } else {
+    $('#name-error').hide();
+  }
+
+  if ($("#hoc-email").val() === "") {
+    $('#email-error').show();
+    return false;
+  } else {
+    $('#email-error').hide();
+  }
+
+  if ($("#country").val() === "") {
+    $('#country-error').show();
+    return false;
+  } else {
+    $('#country-error').hide();
+  }
+
+  if ($("#hoc-event-type").val() === "") {
+    $('#event-type-error').show();
+    return false;
+  } else {
+    $('#event-type-error').hide();
+  }
+
+  if  ($("#hoc-event-type").val() === "in_school" && $("#country").val() === "US") {
+    if ($("#school-name").val() === "") {
+      $('#school-name-error').show();
+      return false;
+    } else {
+      $('#school-name-error').hide();
+    }
+  }
+
+  if  ($("#hoc-event-type").val() === "in_school" && $("#country").val() === "US") {
```

Comment:
best to combine this if statement with the one above, since both have the same condition, by removing this line and the closing brace two lines up